module/zfs/vdev.c

```diff
@@ -1347,6 +1347,10 @@ vdev_open(vdev_t *vd)
 		vd->vdev_max_asize = max_asize;
 		if (vd->vdev_ashift == 0)
 			vd->vdev_ashift = ashift;
+		else
+			if (vd->vdev_ashift < ASHIFT_MIN ||
+			    vd->vdev_ashift > ASHIFT_MAX)
```

Comment:
`vdev_aux_t` hasn't been changed in ~5 years (https://github.com/openzfs/openzfs/blame/master/usr/src/uts/common/sys/fs/zfs.h#L643), i think it's safe to add `VDEV_AUX_BAD_ASHIFT` to it.

I also tried to add just enough code so when we bubble up the error from `vdev_open()` we have sufficient information to generate a new "vdev.bad_ashift" FMA event (instead of a generic "vdev.unknown"):

```
root@debian-8-zfs:~# fallocate -l 64m $TMPDIR/zpool1.dat
root@debian-8-zfs:~# fallocate -l 64m $TMPDIR/zpool2.dat
root@debian-8-zfs:~# zpool create -f $POOLNAME $TMPDIR/zpool1.dat
root@debian-8-zfs:~# zpool events -c
cleared 8 events
root@debian-8-zfs:~# zpool add $POOLNAME -o ashift=14 $TMPDIR/zpool2.dat # <- without userland ashift checks
cannot add to 'testpool': block size out of range or does not match
root@debian-8-zfs:~# zpool events -v
TIME                           CLASS
Mar 14 2017 19:42:41.707571434 ereport.fs.zfs.vdev.bad_ashift
        class = "ereport.fs.zfs.vdev.bad_ashift"
        ena = 0x21942bacc800001
        detector = (embedded nvlist)
                version = 0x0
                scheme = "zfs"
                pool = 0xd4f8c6bab1189a81
                vdev = 0xcdb757669057238d
        (end detector)
        pool = "testpool"
        pool_guid = 0xd4f8c6bab1189a81
        pool_context = 0x0
        pool_failmode = "wait"
        vdev_guid = 0xcdb757669057238d
        vdev_type = "file"
        vdev_path = "/var/tmp/zpool2.dat"
        vdev_ashift = 0xe
        vdev_complete_ts = 0x0
        vdev_delta_ts = 0x0
        vdev_read_errors = 0x0
        vdev_write_errors = 0x0
        vdev_cksum_errors = 0x0
        parent_guid = 0x545994fec1996fa3
        parent_type = "root"
        vdev_spare_paths = 
        vdev_spare_guids = 
        prev_state = 0x7
        time = 0x58c839a1 0x2a2caeea 
        eid = 0x10

Mar 14 2017 19:42:41.707571434 resource.fs.zfs.statechange
        version = 0x0
        class = "resource.fs.zfs.statechange"
        pool_guid = 0xd4f8c6bab1189a81
        pool_context = 0x0
        vdev_guid = 0xcdb757669057238d
        vdev_state = "UNAVAIL" (0x4)
        vdev_path = "/var/tmp/zpool2.dat"
        vdev_laststate = "ONLINE" (0x7)
        time = 0x58c839a1 0x2a2caeea 
        eid = 0x11
```